provider/mem/mem.go

```diff
@@ -41,40 +47,21 @@ type listeningAlerts struct {
 }
 
 // NewAlerts returns a new alert provider.
-func NewAlerts(m types.Marker, intervalGC time.Duration) (*Alerts, error) {
+func NewAlerts(ctx context.Context, m types.Marker, intervalGC time.Duration, l log.Logger) (*Alerts, error) {
+	ctx, cancel := context.WithCancel(ctx)
 	a := &Alerts{
-		alerts:     map[model.Fingerprint]*types.Alert{},
-		marker:     m,
-		intervalGC: intervalGC,
-		stopGC:     make(chan struct{}),
-		listeners:  map[int]listeningAlerts{},
-		next:       0,
+		alerts:    store.NewAlerts(ctx, intervalGC),
+		cancel:    cancel,
+		listeners: map[int]listeningAlerts{},
+		next:      0,
+		logger:    log.With(l, "component", "provider"),
 	}
-	go a.runGC()
-
-	return a, nil
-}
-
-func (a *Alerts) runGC() {
-	for {
-		select {
-		case <-a.stopGC:
-			return
-		case <-time.After(a.intervalGC):
+	a.alerts.SetGCCallback(func(alerts []*types.Alert) {
+		for _, alert := range alerts {
+			m.Delete(alert.Fingerprint())
```

Comment:
Did you intentionally dropped the comment from L69-71? I think it is still relevant.

```go
// As we don't persist alerts, we no longer consider them after
// they are resolved. Alerts waiting for resolved notifications are
// held in memory in aggregation groups redundantly.
```